lib/connection.js

```diff
@@ -310,16 +310,21 @@ Connection.prototype.request = function(request, options, callback) {
 
   // log api usage and its quota
   httpApi.on('response', function(response) {
-    if (response.headers && response.headers["sforce-limit-info"]) {
-      var apiUsage = response.headers["sforce-limit-info"].match(/api\-usage=(\d+)\/(\d+)/);
-      if (apiUsage) {
-        self.limitInfo = {
-          apiUsage: {
-            used: parseInt(apiUsage[1], 10),
-            limit: parseInt(apiUsage[2], 10)
-          }
-        };
-      }
+    if (response.headers) {
+        if (response.headers["sforce-limit-info"]) {
+            var apiUsage = response.headers["sforce-limit-info"].match(/api\-usage=(\d+)\/(\d+)/);
+            if (apiUsage) {
+                self.limitInfo = {
+                    apiUsage: {
+                        used: parseInt(apiUsage[1], 10),
+                        limit: parseInt(apiUsage[2], 10)
+                    }
+                };
+            }
+        }
+        if (options.returnResponseHeaders) {
```

Comment:
return responseHeaders when options.returnResponseHeaders is true.

Reply:
I think this needs to be reworked. First, `self` is the connection, not the request. So you could be dealing with a response that is not from your request. Second, why do we need to set an option to get the response headers? Instead, I would like to see this as part of the httpApi that is passed with the callback.